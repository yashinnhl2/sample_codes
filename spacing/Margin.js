import React from 'react';
import styled from 'styled-components';
import { m, mb, ml, mr, mt, mx, my } from './core';

const MarginStyle = styled.div`
  ${({ inline }) => inline && 'display: inline-block !important;'}
  ${({ all }) => m(all)}
  ${({ horizontal }) => mx(horizontal)}
  ${({ vertical }) => my(vertical)}
  ${({ top }) => mt(top)}
  ${({ right }) => mr(right)}
  ${({ bottom }) => mb(bottom)}
  ${({ left }) => ml(left)}
`;

/**
 * @typedef {import('./core').ViewportProps} ViewportProps
 */

/**
 * @typedef {import('./core').CoreSizes} MarginSize
 */

/**
 * @typedef {object} MarginProps Margin Component Properties
 * @prop {boolean} [inline] Display the container with inline-block
 * @prop {ViewportProps|MarginSize} [all] Margin on all sides
 * @prop {ViewportProps|MarginSize} [vertical] Top and Bottom margin
 * @prop {ViewportProps|MarginSize} [horizontal] Left and Right margin
 * @prop {ViewportProps|MarginSize} [top] Top margin
 * @prop {ViewportProps|MarginSize} [bottom] Bottom margin
 * @prop {ViewportProps|MarginSize} [left] Left margin
 * @prop {ViewportProps|MarginSize} [right] Right margin
 */

/**
 * A container with configurable responsive margins
 * @type {React.FC<MarginProps>}
 * @param {MarginProps} props
 * @returns
 */
const Margin = props => <MarginStyle {...props} />;

export { Margin };
