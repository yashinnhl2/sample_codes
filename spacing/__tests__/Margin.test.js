import React from 'react';
import matchesSnapshot from '@upgrade/ui-utils/utils/assertions/matches-snapshot';
import { Margin } from '../Margin';

describe('Margin', () => {
  it('Should have no style when no props are passed', () => {
    matchesSnapshot(<Margin />);
  });

  it('Should have inline props if inline prop is passed', () => {
    matchesSnapshot(<Margin inline />);
  });

  it('Should have xs to rem margin from all side', () => {
    matchesSnapshot(<Margin all={{ xs: 'xs' }} />);
  });

  it('Should have xs, sm to rem margin from all side', () => {
    matchesSnapshot(<Margin all={{ xs: 'xs', sm: 'md' }} />);
  });

  it('Should have xs, sm to rem margin from right side only', () => {
    matchesSnapshot(<Margin right={{ xs: 'md', sm: 'lg' }} />);
  });

  it('Should have margin top only without any media queries', () => {
    matchesSnapshot(<Margin top="2xl" />);
  });

  it('Should have margin left only without any media queries', () => {
    matchesSnapshot(<Margin left="3xl" />);
  });

  it('Should have margin left only without any media queries, if user pass string', () => {
    matchesSnapshot(<Margin left="lg" />);
  });

  it('Should have no style, if user pass empty string', () => {
    matchesSnapshot(<Margin left="" />);
  });

  it('Should have empty value, if user pass empty object', () => {
    matchesSnapshot(<Margin left={{}} />);
  });
});
