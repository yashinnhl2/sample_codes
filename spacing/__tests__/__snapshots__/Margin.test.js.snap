// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Margin Should have empty value, if user pass empty object 1`] = `
<div
  className=""
/>
`;

exports[`Margin Should have inline props if inline prop is passed 1`] = `
.c0 {
  display: inline-block !important;
}

<div
  className="c0"
/>
`;

exports[`Margin Should have margin left only without any media queries 1`] = `
.c0 {
  margin-left: 4rem;
}

<div
  className="c0"
/>
`;

exports[`Margin Should have margin left only without any media queries, if user pass string 1`] = `
.c0 {
  margin-left: 1.5rem;
}

<div
  className="c0"
/>
`;

exports[`Margin Should have margin top only without any media queries 1`] = `
.c0 {
  margin-top: 3rem;
}

<div
  className="c0"
/>
`;

exports[`Margin Should have no style when no props are passed 1`] = `
<div
  className=""
/>
`;

exports[`Margin Should have no style, if user pass empty string 1`] = `
<div
  className=""
/>
`;

exports[`Margin Should have xs to rem margin from all side 1`] = `
.c0 {
  margin: 0.25rem;
}

<div
  className="c0"
/>
`;

exports[`Margin Should have xs, sm to rem margin from all side 1`] = `
.c0 {
  margin: 0.25rem;
}

@media (min-width:48em) {
  .c0 {
    margin: 1rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Margin Should have xs, sm to rem margin from right side only 1`] = `
.c0 {
  margin-right: 1rem;
}

@media (min-width:48em) {
  .c0 {
    margin-right: 1.5rem;
  }
}

<div
  className="c0"
/>
`;
